### PR TITLE
Generate documentation from json

### PR DESCRIPTION
**Pull request by elbeejay**
This PR switches our documentation of the individual collections to pull from this JSON file: https://climateknowledgeportal.worldbank.org/themes/cckp/data/catalog/tree_all.json 

@g4brielvs if you could take a look and let me know what you think that'd be great. The build works for me locally and we'll see if the GitHub Action works properly in this PR... :crossed_fingers:

This PR splits the collection documentation/data structure out from the example code as the former is auto-generated each time the documentation is built. The data structure pages are long as they expand all of the nested structuring that's revealed step-by-step when selecting data to download via the CCKP web portal. It is searchable/navigable within the page via the page sections sidebar on the right side of the page enabling click-based-navigation down to the level of individual variables for a given collection.

### CODE REVIEW

**Comment by elbeejay** [comment]
@g4brielvs - does the actions workflow not run for PRs? If so please test locally and if it works for both of us then maybe we can merge and hope it works live on the main branch.

**Comment by g4brielvs** [comment]
> @g4brielvs - does the actions workflow not run for PRs? If so please test locally and if it works for both of us then maybe we can merge and hope it works live on the main branch.


@elbeejay Thanks! I was going to send you an example; you were the fastest 🚀. It won't. The workflow is [set up to only be triggered on pushes against `main`](https://github.com/worldbank/climateknowledgeportal/blob/e5f8ee438205d710e93832e650b4c3b32ec8f4f3/.github/workflows/gh-pages.yml#L3).

**Comment by elbeejay** [comment]
True - we might want to consider changing the yml recipe to run all of the stages *except* the deploy stage for pull requests and branches. Not a huge deal, but could help us test the workflow when making bigger changes like this.

**Comment by g4brielvs** [comment]
That's a good point. Also, I was thinking we may want to create a workflow separate from [gh-pages](https://github.com/worldbank/climateknowledgeportal/blob/main/.github/workflows/gh-pages.yml) for the collection documentation.

File: docs/file-structure.md [hunk 1]
```diff
@@ -1,30 +1,20 @@
...
+| COLLECTION LABEL   | FILE STRUCTURE | EXAMPLE |
+|--------------------|----------------|---------|
+| CMIP6 0.25-degree  | [](./collections/cmip6-x0.25.md) | [](../notebooks/cmip6-x0.25.ipynb) |
+| CRU 0.5-degree     | [](./collections/cru-x0.5.md) | [](../notebooks/cru-x0.5.ipynb) |
+| ERA5 0.5-degree    | [](./collections/era5-x0.5.md) | [](../notebooks/era5-x0.5.ipynb) |
+| Population and Poverty | [](./collections/pop-x1.md) | [](../notebooks/pop-x1.ipynb) |
 
-> <font color='#c4c43b'>{collection}</font>/<font color='#c43bc4'>{variable}</font>/<font color='#3bc43b'>{model-run}</font>/{file}.nc
 
-## File Name
```
Comment (g4brielvs) [inline docs/file-structure.md:9]: @elbeejay The example color-coded filename was removed. Wasn't it helpful?

File: docs/scripts/pull_from_json.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+r = requests.get(url)
+data = r.json()
+
+# plan to make an md file for each top level dictionary
+for collection in data.keys():
+    # delete file if it exists
+    if os.path.exists('docs/collections/' + collection + '.md'):
+        os.remove('docs/collections/' + collection + '.md')
+    # create file for writing
+    f = open('docs/collections/' + collection + '.md', 'w')
```
Comment (g4brielvs) [inline docs/scripts/pull_from_json.py:18]: @elbeejay If `docs/collections/` doesn't exist, it will raise an error.

**Review by g4brielvs — CHANGES_REQUESTED** [review]
@elbeejay Thanks! If the proposed changes, it should work. It would look a bit funky though. Are these all the changes the CCPK team requested? Wasn't the color-coded filename example helpful?

File: docs/scripts/pull_from_json.py (hunk 2 above)
Comment (elbeejay) [reply docs/scripts/pull_from_json.py:18]: Right - the aim is to ensure that the documentation is always up-to-date, so by not commiting the subdirectory to the repository and removing it, we can have confidence that if the documentation builds, the latest JSON file was fetched and these pages were generated.

I figured it was a safe way to make sure we didn't end up with stale pages if or as collections are edited and modified.

File: docs/file-structure.md (hunk 1 above)
Comment (elbeejay) [reply docs/file-structure.md:9]: I thought it was nice when all of the subsequent documentation was below and followed the same color scheme. Now that each collection has it's own page I could go either way - if you like it I'm happy to keep it in. Let me know.

**Comment by elbeejay** [comment]
> @elbeejay Thanks! If the proposed changes, it should work. It would look a bit funky though. Are these all the changes the CCPK team requested? Wasn't the color-coded filename example helpful?

What do you mean by "funky"? I would describe this documentation as a bit verbose for sure, but I think the nested indentations for the bullet points makes the structure fairly clear to navigate. The navigation bar on the right side of the page is also convienient to get to the desired timeseries + variable for a given collection.

The CCPK team requested that we split out the documentation to document each collection separately, as the timeseries, variables, models, etc. options are unique for each collection. The idea of automation for the documentation from some sort of web resource was discussed because it sounds like the CCKP collection will be updated, edited, and added to over time, and in general we'd like to avoid having manual updating of the same documentation in multiple places - so that's where the JSON bit came into play.

I made a short comment on the color coding in your inline comment. But thinking about it and looking at it more closely, the other issue is that the color coded structure doesn't seem to capture all of the layers in the heirarchy for these collections. The format `{collection}_{variable}_{model-run}_{file}.nc` doesn't quite capture everything, which for some data like the CMIP6 dataset have `{collection}_{product}_{aggregation}_{time interval}_{percentile}_{scenario}_{model}_{file}.nc` layers to the heirarchy. I also was not 100% sure of how these files are actually named/stored in the data archive, so thought it'd be best to just stick to the self-documenting process from the hosted ("canonical" for our purposes) JSON file.

**Comment by elbeejay** [comment]
Going to note here that via email @g4brielvs said to go ahead and merge these changes.

File: docs/file-structure.md (hunk 1 above)
Comment (g4brielvs) [reply docs/file-structure.md:9]: @elbeejay I was just curious and that makes sense. Thanks!

**Pull request merged**